jenkins/JenkinsfileUpdateEB

```diff
@@ -0,0 +1,375 @@
+*/
+def machineConfigurations = []
+stage('Machine Selection') {
+    node('master') {
+        def machinesToRun = machinesList.findAll({methods.machineCheck(params.machines, it.name)})
+        if (!machinesToRun) {
+            println "No machines were specified. Aborting...."
+            currentBuild.result = "FAILURE"
+            return
+        }
+        for (system in machinesToRun) {
+            machineConfigurations.add(
+                [machine:system, architectures: methods.getMachineConfiguration(
+                        params.machines, system.name, system.archs as String[])])
+        }
+    }
+}
+
+/*
+ Update Stage: configures parallel builds for the different "machinesToRun"
+ Loop over all systems in "machineConfigurations":
+  - if the machine has multiple architectures, starts a sub-loop over architectures
+  - the user parameter "eb_prefix" is prepended to the variable EASYBUILD_PREFIX
+  - EBVERSIONEASYBUILD stores the version of EasyBuild ("params.eb_version")
+  - the target CDT version is selected by the user and stored in "params.cdt_version"
+*/
+def updates = [:]
+def toolchains = "CrayCCE CrayGNU CrayIntel CrayPGI"
+stage('Update Stage') {
+    for (system in machineConfigurations) {
+        def machine = system.machine
+        def architectures = system.architectures
+        def machineName = machine.name
+        for (item in architectures) {
+            /* define architecture and machineLabel
+               to use them in parallel builds */
+            def architecture = item
+            def machineLabel = architecture == "" ?
+               machineName :
+               "$machineName-$architecture"
+            updates["$machineLabel"] = {
+                node(machineName) {
+                    checkout scm
+
+                    /* define command to load EasyBuild */
+                    def load_easybuild = architecture == "" ?
+                        "module load cdt/${params.cdt_version} EasyBuild-custom/cscs" :
+                        "module load cdt/${params.cdt_version} daint-$architecture EasyBuild-custom/cscs"
+
+                    /* define command to switch EasyBuild version*/
+                    def switch_easybuild = params.eb_version == "" ?
+                        "" :
+                        "module switch EasyBuild EasyBuild/${params.eb_version}"
+
+                    /* define list of unuse paths */
+                    def unuse_paths = architecture == '' ?
+                        machine.unusePath.replace(':', ' ') :
+                        machine.unusePath.replace(':', ' ').replace('ARCH', architecture)
+                        println("List of unuse paths: $unuse_paths")
+
+                    /* define eb_flags and command to build software */
+                    def eb_flags = "--ignore-locks -r"
+                    def tc_flags = "--try-toolchain-version=${params.cdt_version}"
+                    //def build_command = architecture == "" ?
```

Comment:
You dan delete this

Reply:
I left it as a comment since we should use the compute nodes, as a general rule. 
However, since currently the build on `scratch` is dramatically slow, I switched to the login node.